### PR TITLE
fix: id:54606,54607 add accessibility label to button for empty butto…

### PR DESCRIPTION
**Pull request by Farhan-emgage**
add accessibility label to button for empty button issue

### CODE REVIEW

**Comment by emgageadmin** [comment]
https://atera.tpondemand.com/entity/54606

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -106,7 +107,10 @@ const button = ({
     : null;
 
   const iconMarkup = icon
-    ? <span className={theme.customIcon}><Icon source={icon} /></span>
+    ? <span className={theme.customIcon}>
+        <Icon source={icon} />
+        {accessibilityLabel && <VisuallyHidden>{accessibilityLabel}</VisuallyHidden> || title && <VisuallyHidden>{title}</VisuallyHidden>}
```
Comment (weegrom) [inline src/components/Button/Button.tsx:112]: @Farhan-emgage this needs a logical check to ensure either `accessibilityLabel` or `title` prop exists. It also needs a logical check to only include the `title` as VisuallyHidden in the case that both exist. Please have an FE team member assist you if needed.

**Review by weegrom — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (Farhan-emgage) [reply src/components/Button/Button.tsx]: @weegrom 

above changes are done

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -105,8 +106,12 @@ const button = ({
     ? <span className={theme.customIcon}><Icon source="caretDown" /></span>
     : null;
 
+    const titleOrAccesibility = title || accessibilityLabel;
   const iconMarkup = icon
-    ? <span className={theme.customIcon}><Icon source={icon} /></span>
+    ? <span className={theme.customIcon}>
+        <Icon source={icon} />
+        {titleOrAccesibility && <VisuallyHidden>{titleOrAccesibility}</VisuallyHidden>}
```
Comment (weegrom) [inline src/components/Button/Button.tsx:113]: @emgage-smit can you review this logic. It needs to:
1. Check if title or accessibilityLabel exists, and if so, place it within the VisuallyHidden component.
2. If both exists then title should be used
3. If nether exist, then it should bypass adding the VisuallyHidden component

I feel the code has a weird structure, as also does not meet requirements 2 and 3.

Can you please review and provide Farhan some assistance if needed. Thanks

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -1035,12 +1039,15 @@ class Pagination extends React.PureComponent<IProps, IState> {
...
     return (
+      <>
+      <VisuallyHidden componentId="pagination_wrap">Pagination</VisuallyHidden>
       <ul
         className={classNames(theme['rc-pagination'], className)}
         style={props.style}
         unselectable="unselectable"
         ref={this.savePaginationNode}
         {...dataOrAriaAttributeProps}
+        aria-labeledby="pagination_wrap"
```
Comment (weegrom) [inline src/components/Pagination/Pagination.tsx:1050]: @Farhan-emgage this property is misspelled. The correct spelling is "aria-labelledby".

Comment (Farhan-emgage) [reply src/components/Pagination/Pagination.tsx:1050]: @weegrom  ok

**Pull request merged**